triton-vm/src/table/processor_table.rs

```diff
@@ -1430,19 +1432,26 @@ impl SingleRowConstraints {
 
 #[derive(Debug, Clone)]
 pub struct RowPairConstraints {
-    variables: [ConstraintCircuitMonad<ProcessorTableChallenges>; 2 * FULL_WIDTH],
-    circuit_builder: ConstraintCircuitBuilder<ProcessorTableChallenges>,
-    zero: ConstraintCircuitMonad<ProcessorTableChallenges>,
-    one: ConstraintCircuitMonad<ProcessorTableChallenges>,
-    two: ConstraintCircuitMonad<ProcessorTableChallenges>,
+    variables: [ConstraintCircuitMonad<ProcessorTableChallenges, DualRowIndicator<FULL_WIDTH>>;
+        2 * FULL_WIDTH],
+    circuit_builder:
+        ConstraintCircuitBuilder<ProcessorTableChallenges, DualRowIndicator<FULL_WIDTH>>,
+    zero: ConstraintCircuitMonad<ProcessorTableChallenges, DualRowIndicator<FULL_WIDTH>>,
+    one: ConstraintCircuitMonad<ProcessorTableChallenges, DualRowIndicator<FULL_WIDTH>>,
+    two: ConstraintCircuitMonad<ProcessorTableChallenges, DualRowIndicator<FULL_WIDTH>>,
 }
 
 impl Default for RowPairConstraints {
     fn default() -> Self {
-        let circuit_builder =
-            ConstraintCircuitBuilder::<ProcessorTableChallenges>::new(2 * FULL_WIDTH);
+        let circuit_builder = ConstraintCircuitBuilder::<
+            ProcessorTableChallenges,
+            DualRowIndicator<FULL_WIDTH>,
+        >::new(2 * FULL_WIDTH);
         let variables_as_circuits = (0..2 * FULL_WIDTH)
-            .map(|i| circuit_builder.input(i))
+            .map(|i| {
+                let input: DualRowIndicator<FULL_WIDTH> = i.into();
+                circuit_builder.input(input)
```

Comment:
The type can be inferred automatically.
```suggestion
                circuit_builder.input(i.into())
```